src/Agent.Plugins/PipelineArtifact/FileContainerProvider.cs

```diff
@@ -0,0 +1,175 @@
+﻿using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.FileContainer;
+using Microsoft.VisualStudio.Services.FileContainer.Client;
+using Microsoft.VisualStudio.Services.WebApi;
+using Minimatch;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Agent.Plugins.PipelineArtifact
+{
+    class FileContainerProvider : IArtifactProvider
+    {
+        private FileContainerHttpClient containerClient;
+        // https://github.com/Microsoft/azure-pipelines-task-lib/blob/master/node/docs/findingfiles.md#matchoptions
+        private static readonly Options minimatchOptions = new Options
+        {
+            Dot = true,
+            NoBrace = true,
+            NoCase = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? true : false
+        };
+
+        public FileContainerProvider(VssConnection connection)
+        {
+            containerClient = connection.GetClient<FileContainerHttpClient>();
+        }
+        public async Task DownloadSingleArtifactAsync(PipelineArtifactDownloadParameters downloadParameters, BuildArtifact buildArtifact, CancellationToken cancellationToken)
+        {
+            await this.DownloadFileContainerAsync(downloadParameters.ProjectId, buildArtifact, downloadParameters.TargetDirectory, downloadParameters.MinimatchFilters, cancellationToken);
+        }
+
+        public async Task DownloadMultipleArtifactsAsync(PipelineArtifactDownloadParameters downloadParameters, IEnumerable<BuildArtifact> buildArtifacts, CancellationToken cancellationToken)
```

Comment:
@omeshp - Could you please review the FileContainerProvider.cs file. Specifically , if we are using the FCS APIs in a correct way, plus the minimatch pattern. How does the FCS client filters files based on minimatch pattern??

Reply:
Also, do we always need ProjectId to retrieve the FCS container , is there any api using ProjectName?

Reply:
@fadnavistanmay I don't think FCS client supports minimatch filters. Also, couldn't find an API which takes project name as input.